stdlib/public/core/StringIndex.swift

```diff
@@ -168,16 +165,17 @@ extension String.Index {
     self.init(encodedOffset: offset, transcodedOffset: 0)
   }
 
+  /// This constructor used to set the `characterStride` cache, however it was
+  /// removed in Swift 5.7 so this constructor is kept for ABI reasons only
   @usableFromInline
```

Comment:
I recommend marking this entry point as unavailable, so newly built code won't be able to call it:

```suggestion
  /// This constructor used to set the `characterStride` cache, however it was
  /// removed in Swift 5.7 so this constructor is kept for ABI reasons only
  @available(*, unavailable)
  @usableFromInline
```

(The compiler will still emit code for such functions, so any binaries that were built before this change will still be able to call this function.)